specs/core/0_beacon-chain.md

```diff
@@ -1302,26 +1331,21 @@ def process_rewards_and_penalties(state: BeaconState) -> None:
 def process_registry_updates(state: BeaconState) -> None:
     # Process activation eligibility and ejections
     for index, validator in enumerate(state.validators):
-        if (
-            validator.activation_eligibility_epoch == FAR_FUTURE_EPOCH
-            and validator.effective_balance == MAX_EFFECTIVE_BALANCE
-        ):
+        if is_eligible_for_activation_queue(validator):
             validator.activation_eligibility_epoch = get_current_epoch(state)
 
         if is_active_validator(validator, get_current_epoch(state)) and validator.effective_balance <= EJECTION_BALANCE:
             initiate_validator_exit(state, ValidatorIndex(index))
 
-    # Queue validators eligible for activation and not dequeued for activation prior to finalized epoch
+    # Queue validators eligible for activation and not yet dequeued for activation
     activation_queue = sorted([
         index for index, validator in enumerate(state.validators)
-        if validator.activation_eligibility_epoch != FAR_FUTURE_EPOCH
-        and validator.activation_epoch >= compute_activation_exit_epoch(state.finalized_checkpoint.epoch)
+        if is_eligible_for_activation(state, validator)
     ], key=lambda index: state.validators[index].activation_eligibility_epoch)
```

Comment:
Repost: Maybe explicitly point out the tiebreaker is `index` for the readers, and for abstracting the sorting algorithm.

```suggestion
    # Order by the sequence of activation_eligibility_epoch setting and then index.
    ], key=lambda index: (state.validators[index].activation_eligibility_epoch, index))
```